src/cluster_legacy.c

```diff
@@ -3595,6 +3749,20 @@ void clusterBroadcastMessage(clusterMsgSendBlock *msgblock) {
 /* Build the message header. hdr must point to a buffer at least
  * sizeof(clusterMsg) in bytes. */
 static void clusterBuildMessageHdr(clusterMsg *hdr, int type, size_t msglen) {
+    hdr->ver = htons(CLUSTER_PROTO_VER);
+    hdr->sig[0] = 'R';
+    hdr->sig[1] = 'C';
+    hdr->sig[2] = 'm';
+    hdr->sig[3] = 'b';
+    hdr->type = htons(type);
+    hdr->totlen = htonl(msglen);
+
+    if (IS_PUBSUB_LIGHT_MESSAGE(type)) {
```

Comment:
This is a coupling that we should avoid. The function is, supposedly, building a generic light message **header** so it should not be concerned about the actual payload type.

```suggestion
    if (IS_LIGHT_MESSAGE(type)) {
```